contrib/mac/framework/julia-config.jl

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env julia
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+const options =
+[
+    "--cflags",
+    "--ldflags",
+    "--ldlibs"
+];
```

Comment:
I based this on the conventional `julia-config.jl` so it  could be merged back no problem.  I added a `Base.DARWIN_FRAMEWORK` boolean that could be checked in `julia-config.jl` to alter behavior.